### ISSUE TITLE
django-sekizai doesn't work with offline compression

### ISSUE THREAD

**Issue by treyhunner** [django-compressor/django-compressor#351]
The django-sekizai section of the documentation should be updated to either:
1. Note that using django-sekizai in the way [described in the docs](https://django_compressor.readthedocs.org/en/latest/django-sekizai/) doesn't work with offline compression since no compress block is used
2. Explain how offline compression works with django-sekizai (what methods work and which ones do not)

Edit: The incompatibility is now mentioned in the docs in https://github.com/django-compressor/django-compressor/pull/875/commits/fde0e62b4366cf55e395be8eb9ba1d57db441273. this issue is now about the incompatibility itself.

**Comment by Chris7** [comment]
Wish I knew about this earlier, would have saved me a few hours.

**Comment by jrief** [comment]
We now have a solution with offline compression:
https://github.com/jrief/django-sekizai-processors

I'am currently building a version which works without sekizai.

**Comment by Chris7** [comment]
Just a note that jrief's comment seems irrelevant to this conversation to save others time. 
https://github.com/jrief/django-sass-processor/issues/2

**Comment by Chris7** [comment]
I'm not sure about the potential consequences of this, but by checking if a node is None in the remove_block_nodes method within compressor/offline/django.py will allow you to move forward. The None blocks appear to be related to calling {{ block.super }} when there is no super block (which is present even in some dependencies, such as oauth2_provider/templates/oauth2_provider/base.html, so it's difficult to fix since this pattern isn't relegated to the user's project always).

**Comment by bxm156** [comment]
+1 Just ran into this issue as well

**Comment by Mimino666** [comment]
+1

**Comment by karyon** [comment]
has anyone here ever made offline compression work with django-sekizai? as in, is there any workaround that we can document?

**Comment by jrief** [comment]
I don't believe it will every work. Reason is, that pages can be build dynamically using ``{% include "..." %}``s. The intention of sekizai is to allow ``{% addtoblock "..." %}`s inside the included template fragment. Therefore **django-compressor** only "sees" the files to include when the page is actually rendered. This is something which can't be done though static template analysis and thus only during runtime.

**Comment by idogrosberg** [comment]
+1
